framework/console/Controller.php

```diff
@@ -87,14 +87,15 @@ public function isColorEnabled($stream = \STDOUT)
     /**
      * Runs an action with the specified action ID and parameters.
      * If the action ID is empty, the method will use [[defaultAction]].
+     * @param Request $request the request instance.
      * @param string $id the ID of the action to be executed.
      * @param array $params the parameters (name-value pairs) to be passed to the action.
      * @return int the status of the action execution. 0 means normal, other values mean abnormal.
      * @throws InvalidRouteException if the requested action ID cannot be resolved into an action successfully.
      * @throws Exception if there are unknown options or missing arguments
      * @see createAction
      */
-    public function runAction($id, $params = [])
+    public function runAction($request, $id, $params = [])
```

Comment:
Could be type-hinted with `Request`.

Reply:
Same as https://github.com/yiisoft/yii2/pull/15459#discussion_r160115321

Reply:
OK.